pallets/inflation/src/lib.rs

```diff
@@ -0,0 +1,601 @@
+// This file is part of Astar.
+
+// Copyright (C) 2019-2023 Stake Technologies Pte.Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// Astar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Astar is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Astar. If not, see <http://www.gnu.org/licenses/>.
+
+//! TODO
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+use astar_primitives::{Balance, BlockNumber};
+use frame_support::{
+    pallet_prelude::*,
+    traits::{Currency, OnTimestampSet},
+};
+use frame_system::{ensure_root, pallet_prelude::*};
+use sp_runtime::{traits::CheckedAdd, Perquintill, Saturating};
+
+pub mod weights;
+pub use weights::WeightInfo;
+
+#[cfg(any(feature = "runtime-benchmarks"))]
+pub mod benchmarking;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+
+    use super::*;
+
+    #[pallet::pallet]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    // Negative imbalance type of this pallet.
+    pub(crate) type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<
+        <T as frame_system::Config>::AccountId,
+    >>::NegativeImbalance;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config<BlockNumber = BlockNumber> {
+        /// The currency trait.
+        /// This has been soft-deprecated but it still needs to be used here in order to access `NegativeImbalance`
+        // which is defined in the currency trait.
+        type Currency: Currency<Self::AccountId, Balance = Balance>;
+
+        /// Handler for 'per-block' payouts.
+        type PayoutPerBlock: PayoutPerBlock<NegativeImbalanceOf<Self>>;
+
+        /// Cycle ('year') configuration - covers periods, subperiods, eras & blocks.
+        type CycleConfiguration: CycleConfiguration;
+
+        /// The overarching event type.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+        /// Weight information for extrinsics in this pallet.
+        type WeightInfo: WeightInfo;
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(crate) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// Inflation parameters have been force changed. This will have effect on the next inflation recalculation.
+        InflationParametersForceChanged,
```

Comment:
I put this on purpose because I want it to be clear when someone looks at the events, it's clear that it was _forced_.

And the new config can be _force set_, _force_recalculated_ and just _recalculated_. I need to distinguish between those.